### ISSUE TITLE
MDBP: MongoDB Preview Release Epic

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma#7903]
For the Preview release, we'd like to:

- Open MongoDB support up to everyone
- Offer a stable experience for the core features we support

In Preview, you'll be able to use most of the query capabilities of the Prisma Client. See our [API Reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference) for the details.

**Remaining Tasks:** _(due: July 13th)_

- [x] https://github.com/prisma/prisma-engines/pull/2043 **@prisma/client**
  This is important internally to guarantee that Prisma API features like nested writes work as expected with MongoDB. We were waiting on the MongoDB team for driver support. They recently shipped a beta last week, so we're working on integrating that.

- [x] https://github.com/prisma/prisma-engines/issues/1601 **@prisma/client**
   Our initial implementation didn't have many tests. We started porting much of our existing SQL test suite to also test MongoDB.

- [x] https://github.com/prisma/prisma/issues/6717 **@prisma/client**
  This is easy to run into when you're using the VSCode extension and you add your first relation. Right now, it's unclear how to proceed since need to add the native type to the field that's not underlined in red. Fixing this would add a tiny bit of polish to the experience.
- [x] #7951 **@prisma/client**
  Right now the mongodb preview flag is hidden. We'll want VSCode to now recommend it

**Note:** The following can happen in parallel with the work above.

- [x] https://github.com/prisma/docs/pull/1900 **@prisma/docs**
   Finish writing up the Getting Started with MongoDB guide in the docs

- [x] https://github.com/prisma/homepage-v8/pull/299
  Add a link from the homepage to the Getting Started with MongoDB guide

-  [x] https://github.com/prisma/homepage-v8/pull/300
   Add a new banner to the top

**Out of Scope for Preview**

- Embedded Document Support
- Introspection Support
- Migration Support

**TODO: discuss**

- Most other Preview flags don't work right now with Mongo turned on.

**Related**

- Community Issue: https://github.com/prisma/prisma/issues/1277

**Comment by matthewmueller** [comment]
MongoDB was released into Preview in 2.27.0!
